File: 2-Ad-Hoc/1030-A-Lenda-de-Falvious-Josephus/1030.py
def flavious(n, k):
    ans, i = 0, 1
    while i <= n:
        ans = (ans + k) % i
        i += 1
    return ans+1
for i in range(1, int(input()) + 1):
    n, k = map(int, input().split())
    print (f"Case {i}: {flavious(n, k)}")

File: 2-Ad-Hoc/1030-A-Lenda-de-Falvious-Josephus/1030.cpp
#include <iostream>

int flavious(int num, int jump) {
    int i = 1, ans = 0;
    while (i <= num) {
        ans = (ans + jump) % i;
        ++i;
    }
    return ans+1;
}

int main() {
    int len, n, k;
    std::cin>>len;
    for (int i = 1; i <= len; ++i) {
        std::cin>>n>>k;
        std::cout<<"Case "<<i<<": "<<flavious(n, k)<<"\n";
    }

    return 0;
}